registry/storage/driver/gcs/gcs.go

```diff
@@ -752,51 +748,64 @@ func (d *driver) Delete(context context.Context, path string) error {
 		}
 		return nil
 	}
-	err = storageDeleteObject(gcsContext, d.bucket, d.pathToKey(path))
-	if err != nil {
-		if status, ok := err.(*googleapi.Error); ok {
-			if status.Code == http.StatusNotFound {
-				return storagedriver.PathNotFoundError{Path: path}
-			}
-		}
+	err = storageDeleteObject(context, d.bucket, d.pathToKey(path), d.GCS)
+	if err == storage.ErrObjectNotExist {
+		return storagedriver.PathNotFoundError{Path: path}
 	}
 	return err
 }
 
-func storageDeleteObject(context context.Context, bucket string, name string) error {
-	return retry(func() error {
-		return storage.DeleteObject(context, bucket, name)
-	})
+func storageDeleteObject(context context.Context, bucket string, name string, gcs *storage.Client) error {
+	if err := gcs.Bucket(bucket).Object(name).Delete(context); err != nil {
```

Comment:
Do we consider to utilize this condition to avoide race condition?

https://pkg.go.dev/cloud.google.com/go/storage#hdr-Conditions